modules/caddypki/ca.go

```diff
@@ -309,14 +309,19 @@ func (ca CA) loadOrGenIntermediate(rootCert *x509.Certificate, rootKey interface
 func (ca CA) genIntermediate(rootCert *x509.Certificate, rootKey interface{}) (interCert *x509.Certificate, interKey interface{}, err error) {
 	repl := ca.newReplacer()
 
-	rootKeyPEM, err := ca.storage.Load(ca.storageKeyRootKey())
-	if err != nil {
-		return nil, nil, fmt.Errorf("loading root key to sign new intermediate: %v", err)
-	}
-	rootKey, err = pemDecodePrivateKey(rootKeyPEM)
-	if err != nil {
-		return nil, nil, fmt.Errorf("decoding root key: %v", err)
+	// XXX: SA4009: argument rootKey is overwritten before first use
+	// Maybe the intent was to load the rootKey if it isn't provided?
+	if rootKey == nil {
+		rootKeyPEM, err := ca.storage.Load(ca.storageKeyRootKey())
+		if err != nil {
+			return nil, nil, fmt.Errorf("loading root key to sign new intermediate: %v", err)
+		}
+		rootKey, err = pemDecodePrivateKey(rootKeyPEM)
+		if err != nil {
+			return nil, nil, fmt.Errorf("decoding root key: %v", err)
+		}
 	}
```

Comment:
I put a comment in the code on this, but want to surface it here too - the `rootKey` passed into the function was never used before (overwritten before first use) - I made an assumption here on how the behaviour maybe should be, but this bears confirmation...

Reply:
Oh, you know, we could probably just use it directly. A quick code search shows that `genIntermediate()` should always called with a non-nil `rootKey` value. I think we can remove this bit entirely, since leaving it in might cover up a bug.

Reply:
Ok, I'll remove it:

```suggestion
```